packages/app/src/systems/Core/hooks/useTransactionCost.ts

```diff
@@ -0,0 +1,28 @@
+import type { ScriptTransactionRequest } from 'fuels';
+import type { UseQueryOptions } from 'react-query';
+import { useQuery } from 'react-query';
+
+import type { TransactionCost } from '../utils/gas';
+import { emptyTransactionCost, getTransactionCost } from '../utils/gas';
+
+import { useEthBalance } from './useEthBalance';
+
+export function useTransactionCost(
+  queryKey: unknown[],
+  request: () => Promise<ScriptTransactionRequest>,
+  options?: Omit<UseQueryOptions<TransactionCost>, 'queryKey' | 'queryFn'>
+) {
+  const ethBalance = useEthBalance();
+
+  if (Array.isArray(queryKey)) {
+    queryKey.push(ethBalance.formatted);
```

Comment:
if I understood you want to have a cache (because that `queryKey` is necessary) based on the Ethereum balance of the user right? Just a point is, if you want to **disable cache** you can simply set `cacheTime: 0` in the query option.

Reference: https://react-query.tanstack.com/reference/useQuery

If you disable cache for this query, you can also remove the `queryKey` parameter

Reply:
No in this case I was to append ethBalance if it changes the networkFee should reload